consumer/heartbeats_consumer.go

```diff
@@ -0,0 +1,315 @@
+package consumer
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	// "github.com/RedHatInsights/insights-results-aggregator/producer"
+	"github.com/RedHatInsights/insights-results-aggregator/storage"
+	"github.com/rs/zerolog/log"
+
+	// "github.com/RedHatInsights/insights-results-aggregator/types"
+	"github.com/Shopify/sarama"
+
+	"github.com/RedHatInsights/insights-results-aggregator/broker"
+)
+
+// Metadata taken from https://github.com/redhatinsights/insights-ingress-go/internal/validators/types.go
+type Metadata struct {
+	IPAddresses    []string          `json:"ip_addresses,omitempty"`
+	Account        string            `json:"account,omitempty"`
+	OrgID          string            `json:"org_id,omitempty"`
+	InsightsID     string            `json:"insights_id,omitempty"`
+	MachineID      string            `json:"machine_id,omitempty"`
+	SubManID       string            `json:"subscription_manager_id,omitempty"`
+	MacAddresses   []string          `json:"mac_addresses,omitempty"`
+	FQDN           string            `json:"fqdn,omitempty"`
+	BiosUUID       string            `json:"bios_uuid,omitempty"`
+	DisplayName    string            `json:"display_name,omitempty"`
+	AnsibleHost    string            `json:"ansible_host,omitempty"`
+	CustomMetadata map[string]string `json:"custom_metadata,omitempty"`
+	Reporter       string            `json:"reporter"`
+	StaleTimestamp time.Time         `json:"stale_timestamp"`
+	QueueKey       string            `json:"queue_key,omitempty"`
+}
+
+// Request taken from https://github.com/redhatinsights/insights-ingress-go/internal/validators/types.go
+type Request struct {
+	Account     string    `json:"account"`
+	Category    string    `json:"category"`
+	ContentType string    `json:"content_type"`
+	Metadata    Metadata  `json:"metadata"`
+	RequestID   string    `json:"request_id"`
+	Principal   string    `json:"principal"`
+	OrgID       string    `json:"org_id"`
+	Service     string    `json:"service"`
+	Size        int64     `json:"size"`
+	URL         string    `json:"url"`
+	ID          string    `json:"id,omitempty"`
+	B64Identity string    `json:"b64_identity"`
+	Timestamp   time.Time `json:"timestamp"`
+}
+
+// KkfMessageProcessor Processor for kafka messages used by NewKafkaConsumer
+type KkfMessageProcessor interface {
+	HandleMessage(msg *sarama.ConsumerMessage) error
+}
+
+// KfkConsumer ...
+type KfkConsumer struct {
```

Comment:
why do you need to write the consumer from scratch? for other writers we are reusing [this one](https://github.com/RedHatInsights/insights-results-aggregator/blob/8ea82653218e805ceb9398140aad17c854c976d4/consumer/kafka_consumer.go#L61)

Reply:
f.e.: https://github.com/RedHatInsights/insights-results-aggregator/blob/8ea82653218e805ceb9398140aad17c854c976d4/consumer/dvo_rules_consumer.go#L25

Reply:
To be more explicit, as I might be wrong, [KafkaConsumer](https://github.com/RedHatInsights/insights-results-aggregator/blob/master/consumer/kafka_consumer.go#L61) requires a [MessageProcessor](https://github.com/RedHatInsights/insights-results-aggregator/blob/master/consumer/kafka_consumer.go#L65C2-L65C18). Its definition of methods require an [incomingMessage](https://github.com/RedHatInsights/insights-results-aggregator/blob/master/consumer/processing.go#L52) which is different than the ingress incoming message format. I did not know how to reconcile those two.

I feel that extending the incoming struct to support both is not the way, and I couldn't workaround it creating a new type to be one or the other. Plus that will require further changes I believe.

Reply:
ok, ty!